app/controllers/chargeback_controller.rb

```diff
@@ -338,6 +389,56 @@ def cb_assign_field_changed
     end
   end
 
+  # Add a new tier at the end
+  def cb_tier_add
+    @edit = session[:edit]
+    i = params[:id]
+    ii = i.to_i
+    @sb[:num_tiers][ii] =
+        @edit[:new][:details][ii][:chargeback_tiers].to_a.length if @edit[:new][:details][ii][:chargeback_tiers]
+    @sb[:num_tiers][ii] = 1 unless @sb[:num_tiers][ii] || @sb[:num_tiers][ii] == 0
+    @sb[:num_tiers][ii] += 1
+    pos = @sb[:num_tiers][ii] - 2
+    n = pos + 1
+    @edit[:new][:tiers][ii][n] = {}
+    @edit[:new][:tiers][ii][n][:start] = @edit[:new][:tiers][ii][pos][:end]
+    @edit[:new][:tiers][ii][n][:end] = Float::INFINITY
+    @edit[:new][:tiers][ii][n][:fix_rate] = 0.0
+    @edit[:new][:tiers][ii][n][:var_rate] = 0.0
```

Comment:
I have great difficulty understanding what this method is doing.  Can we modify `@sb` `@edit` and render in different well named methods?  It would be great to create side-effect free methods that return hashes and arrays, etc. to pass to other methods that set `@sb` and `@edit`.